lib/rules/one-var.js

```diff
@@ -62,20 +120,46 @@ module.exports = function(context) {
         "FunctionDeclaration": startFunction,
         "FunctionExpression": startFunction,
         "ArrowFunctionExpression": startFunction,
+        "BlockStatement": startBlock,
```

Comment:
A switch statement is a new scope per https://people.mozilla.org/~jorendorff/es6-draft.html#sec-switch-statement-runtime-semantics-evaluation

It looks like this might be a bug in Babel. @sebmck it looks like `switch` should create a new scope but isn't. Can you confirm?